source/storage/apis-1.5.rst

```diff
@@ -432,6 +509,47 @@ Request Headers
     response will be returned.
 
 
+**X-Weave-Records**
+
+    This header may be sent with multi-record uploads, to indicate the
+    total number of records included in the request.  If the server
+    would not accept an upload containing that many records, then a
+    **400 Bad Request** response will be returned with response code **17**.
+
+
+**X-Weave-Bytes**
+
+    This header may be sent with multi-record uploads, to indicate the
+    combined size of payloads in the upload, in bytes.  If the server
+    would not accept an upload containing that many bytes, then a
+    **400 Bad Request** response will be returned with response code **17**.
+
+
+**X-Weave-Batch-Records**
+
+    This header may be included with a POST request using the **batch** query
+    parameter, to indicate the total number of records in the batch.
```

Comment:
Yeah, see note below!

English is a squirming beast — we upload things "in batches of 100", so each collection of records is a batch… but we're also uploading a collection of identical batches, and it's also reasonable to call that collection of batches a batch! That's particularly true from the server's perspective.

And it's also (again, primarily from the server's perspective) fair to call the entire collection of all uploaded records a batch — after all, they'll be processed together in one go once they leave the staging area.

("a quantity or consignment of goods produced at one time: a batch of cookies | the company undertakes thirty-six separate quality control checks on every batch.")

I don't know what word one would use for a sequential collection of batches of items.

Maybe we should always use "batch" for a collection of records uploaded at one time, and "transaction" or something like it for all of the records uploaded to process together? 'Total' or 'transaction' both work for me.